src/models/multitrackmodel.cpp

```diff
@@ -3044,29 +3044,30 @@ void MultitrackModel::moveTrack(int fromTrackIndex, int toTrackIndex)
     // Take a copy of the track
     QScopedPointer<Mlt::Producer> producer(m_tractor->multitrack()->track(fromMltIndex));
 
-    // Save the transition b tracks and calculate the new mix mode indices to be reapplied later.
-    QVector<int> bTracks;
-    QMap<int, QString> modes;
+    // Take a copy of all the transitions
+    // Save the transitions in order of their new track index
+    QMap<int, Mlt::Transition *> videoTransitions;
+    QMap<int, Mlt::Transition *> audioTransitions;
```

Comment:
We should avoid using pointers so much. `Mlt::Transition` has copy constructors from another `Transition` as well as `Service`. This avoids needing to `delete` and the casting:
```
--- src/models/multitrackmodel.cpp
+++ src/models/multitrackmodel.cpp
@@ -3046,8 +3046,8 @@ void MultitrackModel::moveTrack(int fromTrackIndex, int toTrackIndex)
 
     // Take a copy of all the transitions
     // Save the transitions in order of their new track index
-    QMap<int, Mlt::Transition *> videoTransitions;
-    QMap<int, Mlt::Transition *> audioTransitions;
+    QMap<int, Mlt::Transition> videoTransitions;
+    QMap<int, Mlt::Transition> audioTransitions;
     QScopedPointer<Mlt::Service> service(m_tractor->producer());
     while (service && service->is_valid()) {
         if (service->type() == mlt_service_transition_type) {
@@ -3064,9 +3064,9 @@ void MultitrackModel::moveTrack(int fromTrackIndex, int toTrackIndex)
                 }
             }
             if (service->get("mlt_service") != QString("mix") ) {
-                videoTransitions[newBTrack] = new Mlt::Transition((mlt_transition) service->get_service());
+                videoTransitions[newBTrack] = *service;
             } else {
-                audioTransitions[newBTrack] = new Mlt::Transition((mlt_transition) service->get_service());
+                audioTransitions[newBTrack] = *service;
             }
         }
         service.reset(service->producer());
@@ -3108,27 +3108,25 @@ void MultitrackModel::moveTrack(int fromTrackIndex, int toTrackIndex)
     // Add the transitions back in the new positions
     for (auto bTrack : audioTransitions.keys()) {
         int aTrack = 0;
-        Mlt::Transition *aTransition = audioTransitions[bTrack];
-        if (aTransition) {
+        Mlt::Transition aTransition = audioTransitions[bTrack];
+        if (aTransition.is_valid()) {
             m_tractor->plant_transition(aTransition, aTrack, bTrack);
-            delete aTransition;
         }
-        Mlt::Transition *vTransition = videoTransitions[bTrack];
-        if (vTransition) {
+        auto vTransition = videoTransitions[bTrack];
+        if (vTransition.is_valid()) {
             if (bTrack == 1) {
-                vTransition->set("disable", 1);
+                vTransition.set("disable", 1);
             } else {
                 // video transitions mix with track 1
                 // (except track 1 which mixes with track 0)
                 aTrack = 1;
-                vTransition->set("disable", 0);
-                QString mode = vTransition->get("1");
+                vTransition.set("disable", 0);
+                QString mode = vTransition.get("1");
                 if (mode.isEmpty()) {
-                    vTransition->set("1", "normal");
+                    vTransition.set("1", "normal");
                 }
             }
             m_tractor->plant_transition(vTransition, aTrack, bTrack);
-            delete vTransition;
         }
     }
 ```

Reply:
Done